vh-arm-templates/AzureFunction/AzureFunction.json

```diff
@@ -0,0 +1,86 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "webAppPortalName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the function app that you wish to create."
+      }
+    },
+    "appServicePlanName": {
+      "type": "string"
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_RAGRS"
+      ],
+      "metadata": {
+        "description": "Storage Account type"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": {
+    "functionAppName": "[parameters('webAppPortalName')]",
+    "hostingPlanName": "[parameters('appServicePlanName')]",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'functions')]",
```

Comment:
For the storage account name I would have used something like this `[replace(parameters('webAppPortalName'),'-', '')]` but therte would more things to consider as the storage account has a name limitation to 24 char. We could furthermore trim the name but I see this as a throwaway so that is fine for now.